cleanlab/datalab/internal/issue_manager/data_valuation.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    ClassVar,
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix
+
+from cleanlab.datalab.internal.issue_manager import IssueManager
+
+if TYPE_CHECKING:  # pragma: no cover
+    import pandas as pd
+    from cleanlab.datalab.datalab import Datalab
+
+
+class DataValuationIssueManager(IssueManager):
+    """Manages data sample with low valuation."""
+
+    description: ClassVar[
+        str
+    ] = """Examples who won't contribute a lot for a model's training
+        will have lower valuation scores.
+    """
+
+    issue_name: ClassVar[str] = "data_valuation"
+    issue_score_key: ClassVar[str]
+    verbosity_levels: ClassVar[Dict[int, List[str]]] = {
+        0: [],
+        1: [],
+        2: [],
+        3: ["average_data_valuation"],
+    }
+
+    DEFAULT_THRESHOLDS = 1e-6
+
+    def __init__(
+        self,
+        datalab: Datalab,
+        threshold: Optional[float] = None,
+        k: int = 10,
+    ):
+        super().__init__(datalab)
+        self.k = k
+        self.threshold = threshold if threshold is not None else self.DEFAULT_THRESHOLDS
```

Comment:
a formatting tool, like `black` would convert this to:

```suggestion
        self.threshold = (
             threshold 
             if threshold is not None 
             else self.DEFAULT_THRESHOLDS
         )
```

Reply:
@smttsp Excited to see you want to help contribute to this library!  

I suggest first reading through [our developer guide](https://github.com/cleanlab/cleanlab/blob/master/DEVELOPMENT.md) to learn all of our conventions and see what is happening in our CI. You'll see we are already using `black`, `mypy`, and many other things you suggested. They just don't happen to be in this PR, and if you look at [the CI logs](https://github.com/cleanlab/cleanlab/actions/runs/6194792316/job/16818336127?pr=850), you'll see it is complaining about these things already (automatically so reviewers don't have to)

Reply:
@smttsp If you'd like to contribute code, a good place to get started is with a GH issue marked as 'good first issue'

https://github.com/cleanlab/cleanlab/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

Reply:
@jwmueller, thanks for your response. Happy to contribute! 

Sure, I read the developer guide and a bunch of other stuff. It is quite large and great work!

> good first issue

The issues seem to be quite complex and require some good understanding of the repo and the work. But I will play with the repo today.